### ISSUE TITLE
Apt public key for gluster.org changed address

### ISSUE THREAD

**Issue by martialblog** [voxpupuli/puppet-gluster#102]
Tried to use puppet-gluster (with 3.9) on Debian 8 and noticed that the apt repository installation failed because the public key is no longer at: https://download.gluster.org/pub/gluster/glusterfs/LATEST/rsa.pub

For 3.9 it's at: 
https://download.gluster.org/pub/gluster/glusterfs/3.9/rsa.pub

For other versions it's at:
https://download.gluster.org/pub/gluster/glusterfs/VERSION/LATEST/rsa.pub

I'll have a look at the module and (hopefully) provide a fix soon...

## Affected Puppet, Ruby, OS and module versions/distributions
- Puppet: 4.6.0
- Module version: 2.3.0

## How to reproduce (e.g Puppet code you use)
```
class { gluster::install:
  server  => true,
  repo    => true,
}
```
## Output log
`Error: /Stage[main]/Gluster::Repo::Apt/Apt::Source[glusterfs-LATEST]/Apt::Key[Add key: A4703C37D3F4DE7F1819E980FE79BB52D5DC52DC from Apt::Source glusterfs-LATEST]/Apt_key[Add key: A4703C37D3F4DE7F1819E980FE79BB52D5DC52DC from Apt::Source glusterfs-LATEST]/ensure: change from absent to present failed: 404 Not Found for https://download.gluster.org/pub/gluster/glusterfs/LATEST/rsa.pub
`

**Comment by martialblog** [comment]
This worked for me, but I wanna see if there's a better solution: https://github.com/martialblog/puppet-gluster/commit/59ca85c945f1fcf87a37ba89fde600ac89d5803c

Since it's going to probably not work with the next release. The directory structure on gluster.org unfortunately differs a bit for some releases.

Also asked the gluster.org Mailing List what's going on here...

**Comment by martialblog** [comment]
Hi, 

I decided to refator the gluster::repo::apt class. The Key ID and Download URL Parameter are replaced by 'release' which will then set them accordingly. 

Also added tests for this. 

https://github.com/martialblog/puppet-gluster/commit/feaa593436f246b89fc967de6928767a51152e23

I'll rebase before opening a Pull Request.
Feedback would be appreciated. Cheers

**Comment by dhoppe** [comment]
This looks great, could you please create an PR?

**Comment by martialblog** [comment]
Unfortunately no response from the gluster.org Team yet.
http://lists.gluster.org/pipermail/gluster-users/2017-February/029930.html

**Comment by martialblog** [comment]
Fixed with version 3.0.0

I'll look into writing some Beaker acceptance tests so this can be avoided in the future...
